src/main/scala/cromwell/binding/Scatter.scala

```diff
@@ -17,8 +17,12 @@ object Scatter {
  * @param index Index of the scatter block. The index is computed during tree generation to reflect wdl scatter blocks structure.
  * @param item Item which this block is scattering over
  * @param collection Wdl Expression corresponding to the collection this scatter is looping through
+ * @param parent Parent of this scatter
  */
-case class Scatter(index: Int, item: String, collection: WdlExpression, parent: Option[Scope]) extends Scope {
+case class Scatter(index: Int,
+                   item: String,
+                   collection: WdlExpression,
+                   override val parent: Option[Scope]) extends Scope {
```

Comment:
Might be answering the wrong question... but here goes:
- `class Scatter` implements `trait Scope`
- `Scope` has a `parent: Option[Scope]`
- If anyone ever refactors away the `parent`, the compiler will now throw an error stating that this value doesn't override anything in its class hierarchy.
- All of this is squished syntacticly, since `Scatter` is a `case class`, so it ends up here in the constructor.
- It also signals to the reader that `parent` is completing some contract with a super class/trait, while the other implicitly-`val` variables in the constructor are unique to _this_ `case class`.

IMHO it's similar to using `var x` vs `val x` when you're never going to change the value. Yes the former works, but the latter is "correct".